pipeline/common/downloads.py

```diff
@@ -18,19 +18,28 @@
 logger = get_logger(__file__)
```

Comment:
If you add this path to `pyproject.toml` the `[tool.pyright]` `include` section you'll get some type checking on it that gets checked in CI. I've been slowly doing this for files that I touch. You usually need to fix a few things, but it can catch bugs and help ensure things are properly typed.